### PR TITLE
Feature/85763 session time out page

### PR DESCRIPTION
**Pull request by acarbutt**
### Context
Add a session time out page

### Changes proposed in this pull request
Show session timed out page after session is expired
Added an Authorize filter where applicable to re-direct the user to the login page if un authenticated

### Checklist

- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by acarbutt** [comment]
Merge after release

File: Frontend/Filters/NotAuthorizedRedirectToLoginPage.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Frontend.Filters
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public class NotAuthorizedRedirectToLoginPage : AllowAnonymousAttribute, IAuthorizationFilter
```
Comment (StewartBarker) [inline Frontend/Filters/NotAuthorizedRedirectToLoginPage.cs:9]: Nice

File: Frontend/Pages/SessionTimedOut.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public class SessionTimedOut : PageModel
+    {
+        private readonly IConfiguration _configuration;
+
+        public SessionTimedOut(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+        public string ReturnUrl { get; set; }
```
Comment (StewartBarker) [inline Frontend/Pages/SessionTimedOut.cshtml.cs:16]: Not necessary but think you can add [FromQuery] attribute to this property and use model binding - then don't need to get it from Request.Query below

**Review by StewartBarker — APPROVED** [review]

File: Frontend/Pages/SessionTimedOut.cshtml.cs (hunk 2 above)
Comment (acarbutt) [reply Frontend/Pages/SessionTimedOut.cshtml.cs:16]: oh yes, great point

**Pull request merged**